app/javascript/widget/components/Form/PhoneInput.vue

```diff
@@ -0,0 +1,314 @@
+        : `hover:border-black-300 focus:border-black-300 ${this.inputLightAndDarkModeColor} ${this.inputBorderColor}`;
+    },
+    inputBorderColor() {
+      return `${this.$dm('border-black-200', 'dark:border-black-500')}`;
+    },
+    inputLightAndDarkModeColor() {
+      return `${this.$dm('bg-white', 'dark:bg-slate-600')} ${this.$dm(
+        'text-slate-700',
+        'dark:text-slate-50'
+      )}`;
+    },
+    filteredCountriesBySearch() {
+      return this.countries.filter(country => {
+        const { name, dial_code, id } = country;
+        const search = this.searchCountry.toLowerCase();
+        return (
+          name.toLowerCase().includes(search) ||
+          dial_code.toLowerCase().includes(search) ||
+          id.toLowerCase().includes(search)
+        );
+      });
+    },
+    activeCountry() {
+      if (this.activeCountryCode) {
+        return this.countries.find(
+          country => country.id === this.activeCountryCode
+        );
+      }
+      return '';
+    },
+  },
+  methods: {
+    setContextValue(code) {
+      // This function is used to set the context value.
+      // The context value is used to set the value of the phone number field in the pre-chat form.
+      this.context.model = `${code}${this.phoneNumber}`;
+    },
+    onChange(e) {
+      if (e.target.value === '') {
+        // This is to reset the active country code and dial code when the user clears the phone number field.
+        this.activeCountryCode = '';
+        this.activeDialCode = '';
+      }
+      this.phoneNumber = e.target.value;
+      // This function is used to set the context value when the user types in the phone number field.
+      this.setContextValue();
+    },
+    dropdownItem() {
+      // This function is used to get all the items in the dropdown.
+      return Array.from(
+        this.$refs.dropdown.querySelectorAll(
+          'div.country-dropdown div.country-dropdown--item'
+        )
+      );
+    },
+    focusedOrActiveItem(className) {
+      // This function is used to get the focused or active item in the dropdown.
+      return Array.from(
+        this.$refs.dropdown.querySelectorAll(
+          `div.country-dropdown div.country-dropdown--item.${className}`
+        )
+      );
+    },
+    onKeyDownHandler(e) {
+      // Here the code does is to check if the dropdown is open.
+      // If it is open, then it checks if the user has pressed the arrow down key or arrow up key.
+      // If the user has pressed the arrow down key, then it will increment the selectedIndex by 1.
+      // If the user has pressed the arrow up key, then it will decrement the selectedIndex by 1.
+      // If the user has pressed the enter key, then it will select the country.
+      const { showDropdown, filteredCountriesBySearch, onSelectCountry } = this;
+      const { selectedIndex } = this;
+
+      if (showDropdown) {
+        if (hasPressedArrowDownKey(e)) {
```

Comment:
Let us not use the methods like this. We have added a helper method to make this single. 

Please see the implementation here. 
https://github.com/chatwoot/chatwoot/blob/9ff5978d6e83ec753778a21e274497884fb6b247/app/javascript/dashboard/components/widgets/mentions/mentionSelectionKeyboardMixin.js#L26-L36